DataDog/datadog-operator
# update monocontainer config

### What does this PR do?

Currently, the feature to consolidate Agent processes into a single container is configured in the `DatadogAgent` CR as follows:

```
global:
  containerProcessStrategy:
    type: [unprivilegedMultiProcessContainer/singleProcessContainers]
```

This changes the configuration to

```
global:
  containerStrategy: [single/optimized]
```

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* Deploy the Operator
* Deploy a DatadogAgent without a ContainerStrategy, and ensure pods are using the default (`optimized`) strategy.
* Deploy a DatadogAgent that includes
```
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    containerStrategy: optimized
```
and make sure are still using the default strategy.
* Update the DatadogAgent to
```
    containerStrategy: single
```
and make sure the agent pods have only one container.
* Update the DatadogAgent to enable a feature that requires a privileged agent process
```
  features:
    cspm:
      enabled: true
```
and make sure the `containerStrategy` is overridden, and is reverted back to the default.


### Checklist
